app/cafe.py

```diff
@@ -0,0 +1,28 @@
+import datetime
+from app.errors import (NotVaccinatedError,
+                        OutdatedVaccineError,
+                        NotWearingMaskError)
+
+
+class Cafe:
+    def __init__(self, name: str) -> None:
+        self.name = name
+
+    def visit_cafe(self, visitor: dict) -> str:
+        if "vaccine" not in visitor:
+            raise NotVaccinatedError(f"{visitor.get('name', 'Visitor')}"
+                                     f" is not vaccinated.")
+
+        expiration_date = visitor["vaccine"].get("expiration_date")
+        if expiration_date is not None:
+            if isinstance(expiration_date, str):
+                expiration_date = datetime.datetime.strptime(expiration_date, "%Y-%m-%d").date()
+            if expiration_date < datetime.date.today():
+                raise OutdatedVaccineError(f"{visitor.get('name', 'Visitor')}"
+                                           f"'s vaccine is outdated.")
```

Comment:
The code assumes that `expiration_date` is either a valid date object or a string that can be converted to a date. Consider adding a check to ensure `expiration_date` is a valid date or string before attempting to convert it. This will prevent potential errors if `expiration_date` is of an unexpected type.